### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by JBoss925**
Merge promise-based materialize into master! All endpoints can now be 100% async and non-blocking (since the firebase promises are added to the return chain), so backend can serve many requests at the same time!

### CODE REVIEW

File: app/test/runTests.ts [hunk 1]
```diff
@@ -89,13 +89,20 @@ async function runTest() {
...
-    if (!testsOver) {
-      await runTest();
-    }
-  });
+  await currentlyRunningTest.apply(null, [db]);
+  nextTest();
+  if (!testsOver) {
+    await runTest();
+  }
+  // let promise = currentlyRunningTest.apply(null, [db]);
```
Comment (Enochen) [inline app/test/runTests.ts]: seems like we don't need this anymore

Comment (JBoss925) [reply app/test/runTests.ts]: yeah, I'll remove that real quick. There was really no reason to do it in a promise chain for testing, as we're awaiting inside our tests anyway.

Comment (JBoss925) [reply app/test/runTests.ts]: Ok cool, @Enochen should be gone now.

Comment (JBoss925) [reply app/test/runTests.ts]: wait, gotta pull into dev first lmao

Comment (JBoss925) [reply app/test/runTests.ts]: ok @Enochen now we're good

File: app/logging/logger.ts [hunk 2]
```diff
@@ -106,16 +106,18 @@ export let responseBoxLog = (msg: string) => {
...
-      resv.send("Unable to retrieve logs!");
-    } else {
-      resv.send(data);
-    }
+export function getLogs(dbv: any, reqv: Request, resv: Response) : Promise<any> {
+  return new Promise<any>((resolve, reject) => {
+    fs.readFile('logs.txt', { encoding: null, flag: undefined }, (err: any, data: Buffer) => {
+      if (err) {
+        console.log("\nUnable to retrieve logs!");
+        resolve("Unable to retrieve logs!");
```
Comment (Enochen) [inline app/logging/logger.ts]: maybe makes more sense to reject the promise when file error occurs?

**Pull request merged**